### PR TITLE
Refactor Sharepoint Rego Code

### PR DESCRIPTION
**Pull request by Sloane4**
#  #

## 🗣 Description ##

The rego code has been expanded greatly, but: with the addition of Regal (OPA linter), update is style practices, & increase in duplicate code, there is a need for a "clean up".

## 💭 Motivation and context ##

- Not triggering any linter warnings/errors (i.e. line length, some in, identical test names, constant conditions, wildcard & test outside test package)
- Style uniformity with booleans (i.e. `== true`, `== false`, `not ExampleMethod()`,  `ExampleMethod() == true`, and `Method() := true if {}`
- Match util import to file structure `import data.report.utils.NotCheckedDetails` -> `import data.utils.report.NotCheckedDetails`
- Refactor out a general ReportsUtil file for all product's report strings
- Simplify code as much as possible by reducing duplicate code
Closing #633 

## 🧪 Testing ##

- Unit Tests

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: Rego/SharepointConfig.rego [hunk 1]
```diff
@@ -17,49 +27,68 @@ import data.report.utils.ReportDetailsString
...
-    "Commandlet" : ["Get-SPOTenant", "Get-PnPTenant"],
-    "ActualValue" : [Policy.OneDriveSharingCapability],
-    "ReportDetails" : ReportDetailsBoolean(Status),
-    "RequirementMet" : Status
-}] {
-    Policy := input.SPO_tenant[_]
+# OneDriveSharingCapability == 0 Only People In Organization
+# OneDriveSharingCapability == 3 Existing Guests
+# OneDriveSharingCapability == 1 New and Existing Guests
+# OneDriveSharingCapability == 2 Anyone
```
Comment (crutchfield) [inline Rego/SharepointConfig.rego:57]: Just an observation.  Code would be more readable if we assigned these and used variables below. Now I have to go back infile to decode.

**Review by crutchfield — APPROVED** [review]
Reviewed code changes.  Unit test passes. 

image

File: Rego/SharepointConfig.rego (hunk 1 above)
Comment (Sloane4) [reply Rego/SharepointConfig.rego:57]: Are you thinking along the lines of turning the values into constants?
image
image

Comment (crutchfield) [reply Rego/SharepointConfig.rego:57]: yes

**Comment by Sloane4** [comment]
@nanda-katikaneni ready for merge

**Pull request merged**